// @ston-fi/sdk/dist/chunk-M2TUY2RO.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunkWDMLYN2Vcjs = require('./chunk-WDMLYN2V.cjs');


var _chunkYKFJZE36cjs = require('./chunk-YKFJZE36.cjs');


var _chunkYSE6O2GUcjs = require('./chunk-YSE6O2GU.cjs');


var _chunkGXCQSYGXcjs = require('./chunk-GXCQSYGX.cjs');


var _chunkZJQ75LD3cjs = require('./chunk-ZJQ75LD3.cjs');

// src/contracts/dex/v2_2/router/BaseRouterV2_2.ts
var BaseRouterV2_2 = (_class = class extends _chunkYSE6O2GUcjs.BaseRouterV2_1 {
  static __initStatic() {this.version = _chunkZJQ75LD3cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkYKFJZE36cjs.BasePoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkGXCQSYGXcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunkWDMLYN2Vcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.BaseRouterV2_2 = BaseRouterV2_2;
