// @ston-fi/sdk/dist/chunk-OTU62ONX.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunkBJZDBVO5cjs = require('./chunk-BJZDBVO5.cjs');


var _chunkWDMLYN2Vcjs = require('./chunk-WDMLYN2V.cjs');


var _chunkRLVCYUFPcjs = require('./chunk-RLVCYUFP.cjs');


var _chunkGXCQSYGXcjs = require('./chunk-GXCQSYGX.cjs');


var _chunkZJQ75LD3cjs = require('./chunk-ZJQ75LD3.cjs');

// src/contracts/dex/v2_2/router/WCPIRouterV2_2.ts
var WCPIRouterV2_2 = (_class = class extends _chunkRLVCYUFPcjs.WCPIRouterV2_1 {
  static __initStatic() {this.version = _chunkZJQ75LD3cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkBJZDBVO5cjs.WCPIPoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkGXCQSYGXcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunkWDMLYN2Vcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.WCPIRouterV2_2 = WCPIRouterV2_2;
