// @ston-fi/sdk/dist/chunk-CZAECWQQ.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunkYSE6O2GUcjs = require('./chunk-YSE6O2GU.cjs');


var _chunkECZLZ7PAcjs = require('./chunk-ECZLZ7PA.cjs');


var _chunkZJQ75LD3cjs = require('./chunk-ZJQ75LD3.cjs');

// src/contracts/dex/v2_1/router/StableRouterV2_1.ts
var _ton = require('@ton/ton');
var StableRouterV2_1 = (_class = class _StableRouterV2_1 extends _chunkYSE6O2GUcjs.BaseRouterV2_1 {
  static __initStatic() {this.dexType = _chunkZJQ75LD3cjs.DEX_TYPE.Stable}
  static __initStatic2() {this.gasConstants = {
    ..._chunkYSE6O2GUcjs.BaseRouterV2_1.gasConstants,
    swapJettonToJetton: {
      gasAmount: _ton.toNano.call(void 0, "0.329"),
      forwardGasAmount: _ton.toNano.call(void 0, "0.269")
    },
    swapJettonToTon: {
      gasAmount: _ton.toNano.call(void 0, "0.329"),
      forwardGasAmount: _ton.toNano.call(void 0, "0.269")
    },
    swapTonToJetton: {
      forwardGasAmount: _ton.toNano.call(void 0, "0.329")
    }
  }}
  constructor(address, { gasConstants, ...options } = {}) {
    super(address, {
      ...options,
      gasConstants: {
        ..._StableRouterV2_1.gasConstants,
        ...gasConstants
      }
    });
  }
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkECZLZ7PAcjs.StablePoolV2_1.create(poolAddress);
  }
}, _class.__initStatic(), _class.__initStatic2(), _class);



exports.StableRouterV2_1 = StableRouterV2_1;
