// @ston-fi/sdk/dist/chunk-UFVKG5I6.js
import {
  BaseRouterV2_2
} from "./chunk-W5BTFHCU.js";
import {
  CPIRouterV2_2
} from "./chunk-PGCWH6QI.js";
import {
  StableRouterV2_2
} from "./chunk-GHJFY7T3.js";
import {
  StablePoolV2_2
} from "./chunk-X3PT4LB6.js";
import {
  WCPIRouterV2_2
} from "./chunk-OW5D6WMM.js";
import {
  WCPIPoolV2_2
} from "./chunk-UBOSEJWA.js";
import {
  WStableRouterV2_2
} from "./chunk-5L3VBOB3.js";
import {
  WStablePoolV2_2
} from "./chunk-DCJ3CQ52.js";
import {
  VaultV2_2
} from "./chunk-R6GQHPPW.js";
import {
  BasePoolV2_2
} from "./chunk-UHNXDBA4.js";
import {
  CPIPoolV2_2
} from "./chunk-MVLUFHSB.js";
import {
  LpAccountV2_2
} from "./chunk-XBZ6KIKA.js";
import {
  PtonV2_1
} from "./chunk-4OOOZ56G.js";

// src/contracts/dex/v2_2/index.ts
var RouterV2_2 = class extends BaseRouterV2_2 {
  static CPI = CPIRouterV2_2;
  static Stable = StableRouterV2_2;
  static WCPI = WCPIRouterV2_2;
  static WStable = WStableRouterV2_2;
};
var PoolV2_2 = class extends BasePoolV2_2 {
  static CPI = CPIPoolV2_2;
  static Stable = StablePoolV2_2;
  static WCPI = WCPIPoolV2_2;
  static WStable = WStablePoolV2_2;
};
var DEX = {
  Router: RouterV2_2,
  Pool: PoolV2_2,
  LpAccount: LpAccountV2_2,
  Vault: VaultV2_2,
  pTON: PtonV2_1
};

export {
  RouterV2_2,
  PoolV2_2,
  DEX
};
