// @ston-fi/sdk/dist/chunk-LY3QG3AK.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class; var _class2;

var _chunkM2TUY2ROcjs = require('./chunk-M2TUY2RO.cjs');


var _chunkT3XYWBPScjs = require('./chunk-T3XYWBPS.cjs');


var _chunkF5WFTR4Icjs = require('./chunk-F5WFTR4I.cjs');


var _chunk3TM5O6LUcjs = require('./chunk-3TM5O6LU.cjs');


var _chunkOTU62ONXcjs = require('./chunk-OTU62ONX.cjs');


var _chunkBJZDBVO5cjs = require('./chunk-BJZDBVO5.cjs');


var _chunkPYYA5DMDcjs = require('./chunk-PYYA5DMD.cjs');


var _chunkKNOL2G5Lcjs = require('./chunk-KNOL2G5L.cjs');


var _chunkWDMLYN2Vcjs = require('./chunk-WDMLYN2V.cjs');


var _chunkYKFJZE36cjs = require('./chunk-YKFJZE36.cjs');


var _chunkYV2YJEPScjs = require('./chunk-YV2YJEPS.cjs');


var _chunkG66DMSMEcjs = require('./chunk-G66DMSME.cjs');


var _chunkJEI7ZWK3cjs = require('./chunk-JEI7ZWK3.cjs');

// src/contracts/dex/v2_2/index.ts
var RouterV2_2 = (_class = class extends _chunkM2TUY2ROcjs.BaseRouterV2_2 {
  static __initStatic() {this.CPI = _chunkT3XYWBPScjs.CPIRouterV2_2}
  static __initStatic2() {this.Stable = _chunkF5WFTR4Icjs.StableRouterV2_2}
  static __initStatic3() {this.WCPI = _chunkOTU62ONXcjs.WCPIRouterV2_2}
  static __initStatic4() {this.WStable = _chunkPYYA5DMDcjs.WStableRouterV2_2}
}, _class.__initStatic(), _class.__initStatic2(), _class.__initStatic3(), _class.__initStatic4(), _class);
var PoolV2_2 = (_class2 = class extends _chunkYKFJZE36cjs.BasePoolV2_2 {
  static __initStatic5() {this.CPI = _chunkYV2YJEPScjs.CPIPoolV2_2}
  static __initStatic6() {this.Stable = _chunk3TM5O6LUcjs.StablePoolV2_2}
  static __initStatic7() {this.WCPI = _chunkBJZDBVO5cjs.WCPIPoolV2_2}
  static __initStatic8() {this.WStable = _chunkKNOL2G5Lcjs.WStablePoolV2_2}
}, _class2.__initStatic5(), _class2.__initStatic6(), _class2.__initStatic7(), _class2.__initStatic8(), _class2);
var DEX = {
  Router: RouterV2_2,
  Pool: PoolV2_2,
  LpAccount: _chunkG66DMSMEcjs.LpAccountV2_2,
  Vault: _chunkWDMLYN2Vcjs.VaultV2_2,
  pTON: _chunkJEI7ZWK3cjs.PtonV2_1
};





exports.RouterV2_2 = RouterV2_2; exports.PoolV2_2 = PoolV2_2; exports.DEX = DEX;
