// @ston-fi/sdk/dist/chunk-PYYA5DMD.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunkKNOL2G5Lcjs = require('./chunk-KNOL2G5L.cjs');


var _chunkWDMLYN2Vcjs = require('./chunk-WDMLYN2V.cjs');


var _chunkT2VEERB5cjs = require('./chunk-T2VEERB5.cjs');


var _chunkGXCQSYGXcjs = require('./chunk-GXCQSYGX.cjs');


var _chunkZJQ75LD3cjs = require('./chunk-ZJQ75LD3.cjs');

// src/contracts/dex/v2_2/router/WStableRouterV2_2.ts
var WStableRouterV2_2 = (_class = class extends _chunkT2VEERB5cjs.WStableRouterV2_1 {
  static __initStatic() {this.version = _chunkZJQ75LD3cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkKNOL2G5Lcjs.WStablePoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkGXCQSYGXcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunkWDMLYN2Vcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.WStableRouterV2_2 = WStableRouterV2_2;
