// @ston-fi/sdk/dist/chunk-C2DDULA7.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class; var _class2;

var _chunk235HRTDLcjs = require('./chunk-235HRTDL.cjs');


var _chunkCZAECWQQcjs = require('./chunk-CZAECWQQ.cjs');


var _chunkRLVCYUFPcjs = require('./chunk-RLVCYUFP.cjs');


var _chunkT2VEERB5cjs = require('./chunk-T2VEERB5.cjs');


var _chunk5MJFHJMBcjs = require('./chunk-5MJFHJMB.cjs');


var _chunkYSE6O2GUcjs = require('./chunk-YSE6O2GU.cjs');


var _chunkPDIKU2S5cjs = require('./chunk-PDIKU2S5.cjs');


var _chunkXOHVHVTTcjs = require('./chunk-XOHVHVTT.cjs');


var _chunkECZLZ7PAcjs = require('./chunk-ECZLZ7PA.cjs');


var _chunkYXKUTOAFcjs = require('./chunk-YXKUTOAF.cjs');


var _chunkAJ2JDNJZcjs = require('./chunk-AJ2JDNJZ.cjs');


var _chunkYLEQB4MTcjs = require('./chunk-YLEQB4MT.cjs');


var _chunkJEI7ZWK3cjs = require('./chunk-JEI7ZWK3.cjs');

// src/contracts/dex/v2_1/index.ts
var RouterV2_1 = (_class = class extends _chunkYSE6O2GUcjs.BaseRouterV2_1 {
  static __initStatic() {this.CPI = _chunk235HRTDLcjs.CPIRouterV2_1}
  static __initStatic2() {this.Stable = _chunkCZAECWQQcjs.StableRouterV2_1}
  static __initStatic3() {this.WCPI = _chunkRLVCYUFPcjs.WCPIRouterV2_1}
  static __initStatic4() {this.WStable = _chunkT2VEERB5cjs.WStableRouterV2_1}
}, _class.__initStatic(), _class.__initStatic2(), _class.__initStatic3(), _class.__initStatic4(), _class);
var PoolV2_1 = (_class2 = class extends _chunkAJ2JDNJZcjs.BasePoolV2_1 {
  static __initStatic5() {this.CPI = _chunkXOHVHVTTcjs.CPIPoolV2_1}
  static __initStatic6() {this.Stable = _chunkECZLZ7PAcjs.StablePoolV2_1}
  static __initStatic7() {this.WCPI = _chunkYXKUTOAFcjs.WCPIPoolV2_1}
  static __initStatic8() {this.WStable = _chunk5MJFHJMBcjs.WStablePoolV2_1}
}, _class2.__initStatic5(), _class2.__initStatic6(), _class2.__initStatic7(), _class2.__initStatic8(), _class2);
var DEX = {
  Router: RouterV2_1,
  Pool: PoolV2_1,
  LpAccount: _chunkYLEQB4MTcjs.LpAccountV2_1,
  Vault: _chunkPDIKU2S5cjs.VaultV2_1,
  pTON: _chunkJEI7ZWK3cjs.PtonV2_1
};





exports.RouterV2_1 = RouterV2_1; exports.PoolV2_1 = PoolV2_1; exports.DEX = DEX;
