// @ston-fi/sdk/dist/chunk-5LPE44J5.js
import {
  CPIRouterV2_1
} from "./chunk-IFMGU6GB.js";
import {
  StableRouterV2_1
} from "./chunk-VFE4LTGM.js";
import {
  WCPIRouterV2_1
} from "./chunk-FU2GQY36.js";
import {
  WStableRouterV2_1
} from "./chunk-KTS34XTO.js";
import {
  WStablePoolV2_1
} from "./chunk-D5R4VQ3E.js";
import {
  BaseRouterV2_1
} from "./chunk-YSYMABVC.js";
import {
  VaultV2_1
} from "./chunk-ZREZXXSX.js";
import {
  CPIPoolV2_1
} from "./chunk-PFDB63A3.js";
import {
  StablePoolV2_1
} from "./chunk-KXPOQKNK.js";
import {
  WCPIPoolV2_1
} from "./chunk-XEXEDC3T.js";
import {
  BasePoolV2_1
} from "./chunk-DFPKM3LH.js";
import {
  LpAccountV2_1
} from "./chunk-NYDP354Z.js";
import {
  PtonV2_1
} from "./chunk-4OOOZ56G.js";

// src/contracts/dex/v2_1/index.ts
var RouterV2_1 = class extends BaseRouterV2_1 {
  static CPI = CPIRouterV2_1;
  static Stable = StableRouterV2_1;
  static WCPI = WCPIRouterV2_1;
  static WStable = WStableRouterV2_1;
};
var PoolV2_1 = class extends BasePoolV2_1 {
  static CPI = CPIPoolV2_1;
  static Stable = StablePoolV2_1;
  static WCPI = WCPIPoolV2_1;
  static WStable = WStablePoolV2_1;
};
var DEX = {
  Router: RouterV2_1,
  Pool: PoolV2_1,
  LpAccount: LpAccountV2_1,
  Vault: VaultV2_1,
  pTON: PtonV2_1
};

export {
  RouterV2_1,
  PoolV2_1,
  DEX
};
