// @ston-fi/sdk/dist/chunk-F5WFTR4I.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunk3TM5O6LUcjs = require('./chunk-3TM5O6LU.cjs');


var _chunkWDMLYN2Vcjs = require('./chunk-WDMLYN2V.cjs');


var _chunkCZAECWQQcjs = require('./chunk-CZAECWQQ.cjs');


var _chunkGXCQSYGXcjs = require('./chunk-GXCQSYGX.cjs');


var _chunkZJQ75LD3cjs = require('./chunk-ZJQ75LD3.cjs');

// src/contracts/dex/v2_2/router/StableRouterV2_2.ts
var StableRouterV2_2 = (_class = class extends _chunkCZAECWQQcjs.StableRouterV2_1 {
  static __initStatic() {this.version = _chunkZJQ75LD3cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunk3TM5O6LUcjs.StablePoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkGXCQSYGXcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunkWDMLYN2Vcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.StableRouterV2_2 = StableRouterV2_2;
